src/classes/BDI_DataImportService_TEST.cls

```diff
@@ -0,0 +1,142 @@
+        Data_Import_Settings__c diSettings = getDataImportSettings();
+        diSettings.Run_Opportunity_Rollups_while_Processing__c = true;
+
+        Boolean isDryRun = false;
+        BDI_DataImportService service = new BDI_DataImportService(isDryRun);
+        service.injectDataImportSettings(diSettings);
+
+        List<Trigger_Handler__c> expectedHandlers = TDTM_Config_API.getCachedRecords();
+
+        service.disableAllOppRollups();
+
+        System.assertEquals(expectedHandlers, TDTM_Config_API.getCachedRecords());
+    }
+
+    /*******************************************************************************************************************
+    * @description Verifies rollup trigger handlers are deactivated if the data import settings to process rollups
+    * is not checked.
+    */
+    @isTest
+    private static void rollupTriggersAreDeactivatedWhenRunRollupSettingsIsNotChecked() {
+        Data_Import_Settings__c diSettings = getDataImportSettings();
+        diSettings.Run_Opportunity_Rollups_while_Processing__c = false;
+
+        Boolean isDryRun = false;
+        BDI_DataImportService service = new BDI_DataImportService(isDryRun);
+        service.injectDataImportSettings(diSettings);
+
+        Map<String, Trigger_Handler__c> handlerByKey = new Map<String, Trigger_Handler__c>();
+        for (Trigger_Handler__c handler : TDTM_Config_API.getCachedRecords()) {
+            handlerByKey.put(handler.Object__c + handler.Class__c, handler);
+        }
+
+        service.disableAllOppRollups();
+
+        for (Trigger_Handler__c handler : TDTM_Config_API.getCachedRecords()) {
+            Boolean isActive = isRollupTriggerHandler(handler) 
+                ? false 
+                : handlerByKey.get(handler.Object__c + handler.Class__c).Active__c;
+
+            System.assertEquals(isActive, handler.Active__c, 'Only rollup trigger handlers should be deactivated: ' + handler);
+        }
+    }
+
+
+    // Helpers
+    ////////////
+
+    /*******************************************************************************************************************
+    * @description Retrieves and sets default on the data import settings
+    */
+    private static Data_Import_Settings__c getDataImportSettings() {
+        Data_Import_Settings__c diSettings = UTIL_CustomSettingsFacade.getDataImportSettings();
+
+        diSettings.Donation_Matching_Rule__c = UTIL_Namespace.StrTokenNSPrefix('Donation_Amount__c') + ';' +
+            UTIL_Namespace.StrTokenNSPrefix('Donation_Date__c');
+        diSettings.Donation_Matching_Behavior__c = BDI_DataImport_API.ExactMatchOrCreate;
+
+        return diSettings;
+    }
+
+
+    /*******************************************************************************************************************
+    * @description Returns true if the trigger handler calculates rollups
+    */
+    private static Boolean isRollupTriggerHandler(Trigger_Handler__c handler) {
```

Comment:
If the `TDTM_Config_API.disableAllRollupTriggers()` method is used to disable all the rollup triggers, this method will likely need to be modified to match the same logic.

Reply:
The helper test method does not need to be changed since it specifically verifies which trigger handlers should be deactivated. If the `TDTM_Config_API.disableAllRollupTriggers()` is changed to remove one of those triggers or add new, the `BDI_DataImportService_TEST` will fail, so developer will know additional changes and unit/QA tests need to be done.